File: src/main/java/io/pageobject/generator/locator/protractor/LinkTextLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.Expressions.hasExpression;

public class LinkTextLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        String text = element.text();
        return hasExpression(text) ? null : text;
    }

    public String locatorTemplate() {
        return "by.linkText('%s')";
    }

}

File: lib/locator/protractor/link-text-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const hasExpression = require('../../expressions').hasExpression;
const text = require('../../dom-helpers').text;

const template = _.template("by.linkText('<%= value %>')");

class LinkTextLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    let elementText = text(context.element);
    return hasExpression(elementText) ? null : elementText;
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = LinkTextLocatorSource;